### DISCUSSION TITLE
Documentation on color_mode seems to be missing

### DISCUSSION THREAD

**Discussion by itavero** [Koenkk/zigbee2mqtt#15512] [General]
I noticed that Zigbee2MQTT sends a `color_mode` for lights since quite some time, but I can't seem to find any documentation on it.
It is also not part of the `exposes` information, so I'm wondering what would be the best way to detect if Zigbee2MQTT will publish this information (I want to know this for itavero/homebridge-z2m#488 / itavero/homebridge-z2m#30).

Is there a specific version where this was introduced and will it always be send or does it also require a specific configuration?

**Comment by Koenkk** [comment]
The color mode is present when the light supports one of the following color features: `color_temp`, `xy` (= color in x/y), `hs` (= color in hue/saturation)

**Reply by itavero** [reply]
Thanks. I can't remember it always being there, but I'll have a look if it was already present in Zigbee2MQTT v1.17.0 (which is the oldest version currently supported by homebridge-z2m).

**Reply by sjorge** [reply]
It’s rather useless for homekit, if Hue/Sat is supported you cant send colorTemperature or it will complain :(

I tried sending the correct one to skip passing through the conversions, but you unfortunetly always have to do them. (or take the one z2m does)

Also color_mode is always send, even when the bulb does not report it, z2m will still set/send it.

**Reply by itavero** [reply]
The docs for HAP-NodeJS explicitly mention something about doing a conversion when you support both Color Temperature and Hue/Saturation: https://developers.homebridge.io/HAP-NodeJS/classes/AdaptiveLightingController.html

But indeed, it's a bit weird how HomeKit deals with these devices.

**Reply by sjorge** [reply]
I don’t think the color mode mentioned there refers to the zigbee color mode, but the whole explanation is rather confusing.

**Reply by itavero** [reply]
What they are referring to is that, as a HomeKit accessory that supports both Hue/Saturation and Color Temperature, you have to set the value of the Hue/Saturation to something "similar" to the Color Temperature (for which conversion methods exist).
Based on the `color_mode` from Zigbee2MQTT, I can determine which value from the MQTT message I need to use and if I need to perform the conversion or not.
I've noticed logic for this in other plugins (e.g. homebridge-hue) as well.

If you don't do this, the UI in the Home.app seems to look a bit weird, if I recall correctly.

**Reply by sjorge** [reply]
If color sync is enable in z2m (default) you should be able to just use either color_temp or xy or hs propeties. Depending on what the bulb uses.

z2m will keep these values in sync unless told not to do so: https://github.com/Koenkk/zigbee-herdsman-converters/blob/af0449f85afb747c9cc8ad46c4343d9a1cb98b43/lib/color.js#L647
